src/main/java/io/anserini/collection/TwentynewsgroupsCollection.java

```diff
@@ -0,0 +1,146 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.collection;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.NoSuchElementException;
+
+/**
+ * An instance of the <a href="http://people.csail.mit.edu/jrennie/20Newsgroups">20 Newsgroup by date</a>.
+ * The collection contains 18846 documents, 20 different news sets,each corresponding to a different topic. 
+ * stored in plain text format. The collection is 14MB compressed, 35.9MB uncompressed.
+ */
+public class TwentynewsgroupsCollection extends DocumentCollection<TwentynewsgroupsCollection.Document> {
+
+  public TwentynewsgroupsCollection(Path path){
+    this.path = path;
+  }
+
+  @Override
+  public FileSegment<TwentynewsgroupsCollection.Document> createFileSegment(Path p) throws IOException {
+    return new Segment(p);
+  }
+
+  /**
+   * A file containing one document from the <a href="http://people.csail.mit.edu/jrennie/20Newsgroups">20 Newsgroup by date</a>.
+   * The file Name which is the Id of that document.
+   */
+  public static class Segment extends FileSegment<Document> {
+    private String fileName;
+    private String ID;
+
+    public Segment(Path path) throws IOException {
+      super(path);
+      this.fileName = path.toString();
+      String[] str_path = fileName.split("/");
+      this.ID = str_path[str_path.length-1];
+      this.bufferedReader = new BufferedReader(new InputStreamReader(new FileInputStream(this.fileName), "utf-8"));
+    }
+
+    @Override
+    public void readNext() throws IOException {
+      String record;
+      String From = "";
+      String Subject = "";
+      String Keywords = "";
+      String Organization = "";
+      String s = "";
+      while ((record = bufferedReader.readLine())!=null){
+            String parts[] = record.split(" ", 2);
+            if (record.startsWith("From: ")){     
+                From = parts[1];
+            }else if (record.startsWith("Subject: ")){
+                Subject = parts[1];
+            }else if (record.startsWith("Keywords: ")){
+                Keywords = parts[1];
+            }else if (record.startsWith("Organization: ")){
+                Organization = parts[1];
+            }
+            if (s!=""){
+                s = s+"\n"+record;
+            }else{
+                s = record;
+            }
+      }
+      if (s == "") {
+            throw new NoSuchElementException();
+      }
+      this.bufferedRecord = new TwentynewsgroupsCollection.Document(this.ID,From,Subject,Keywords,Organization,s);
+    }
+
+  }
+  /**
+   * A document from the <a href="http://people.csail.mit.edu/jrennie/20Newsgroups">20 Newsgroup by date</a>.
```

Comment:
what does "by date" mean?

Reply:
For the 20 Newsgroup data set, there're in total three version of it. One of them is called 20newsgroup by date which is recommended to use for this collection, so I commented it there.

Reply:
Ah, I see. Please explain?

Reply:
It might be a better idea to just explain it in a full sentence, so I removed the  'by date' and added a full sentence for explanation in the comments part.